docs/data-formats.md

```diff
@@ -224,22 +224,107 @@ are applied.
 
 ## Custom Attributes Annotations Format
 
-```json
-{
-  "[assembly-name]": {
-    "[namespace-name]": {
-      "[type-name]": {
-        "[field-or-property-name]": {
-          "[attribute-name]": "[attribute-value]"
-        },
-
-        "[method-name]([method-signature])": {
-          "[parameter-name]": {
-            "[attribute-name]": "[attribute-value]"
-          }
-        }
-      }
-    }
-  }
-}
+### Custom attribute on type field
+
+```xml
+<linker>
+  <assembly fullname="Assembly">
+    <type fullname="Assembly.A">
+      <field name="MyNumericField">
+        <attribute fullname="System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers">
+          <argument>DefaultConstructor</argument>
+        </attribute>
```

Comment:
Please add an example with more than 1 argument. We should also document how are the values converted (e.g. how to do pass enum value, a character, null value, etc.)

Reply:
I'm hesitant to try to design the solution for any attribute right now, without having a use case for it. I personally would prefer to only support the necessary minimum we need right now. As we add more attributes we can extend the format as appropriate.

Reply:
That would be a breaking change as I don't think we want to have different syntax for single argument and multiple arguments

Reply:
I don't see how that's the case. I was assuming that if we need more arguments we would simply accept multiple `argument` elements here - order dependent. If we need fields/properties in the future that will need to also specify names.

Reply:
I guess we could document what we think multiple arguments, fields and properties could look like, but I don't think we need to implement all of it right now.

Reply:
I was thinking about something like

```xml
        <attribute fullname="System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers">
		  <arguments>
	          <argument>DefaultConstructor</argument>
		  <arguments>
        </attribute>
```

or 

```xml
        <attribute fullname="System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers">
		  <fields>
	          <field name="Name">SomeValue</field>
		  <field>
        </attribute>
```

Reply:
I don't see a reason to add another nesting. From XML's point of view the `arguments` doesn't really add much value. The only downside is that could write:
```XML
        <attribute fullname="System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers">
          <argument>DefaultConstructor</argument>
          <field name="Name">SomeValue<field>
          <argument>SecondArgumentValue</argument>
        </attribute>
```

Which would work, but it's not very nice, but that's just about writing nice code - I don't think we have to enforce it by the format itself.

Reply:
No problem with that but I think it'd be helpful to document the intention